diff --git a/CMakeLists.txt b/CMakeLists.txt
index f122158..1f97865 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,7 +36,7 @@ execute_process(
 #
 
 include_directories(.)
-add_compile_options(-std=c++23 -DWLR_USE_UNSTABLE )
+add_compile_options(-std=c++2b -DWLR_USE_UNSTABLE )
 add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wno-narrowing -Wno-pointer-arith)
 find_package(Threads REQUIRED)
