diff --git a/src/gui/views/list.rs b/src/gui/views/list.rs
index ddcf1d9..3b58dad 100644
--- a/src/gui/views/list.rs
+++ b/src/gui/views/list.rs
@@ -157,7 +157,7 @@ impl List {
                             );
                             self.selection
                                 .selected_packages
-                                .drain_filter(|s_i| *s_i == i);
+                                .extract_if(|s_i| *s_i == i);
                         }
                     } else if !self.selection.selected_packages.contains(&i) {
                         self.selection.selected_packages.push(i);
@@ -209,7 +209,7 @@ impl List {
                             } else {
                                 self.selection
                                     .selected_packages
-                                    .drain_filter(|s_i| *s_i == i_package);
+                                    .extract_if(|s_i| *s_i == i_package);
                             }
                             update_selection_count(
                                 &mut self.selection,
@@ -269,12 +269,12 @@ impl List {
 
                 match action {
                     Action::Remove => {
-                        selected_packages.drain_filter(|i| {
+                        selected_packages.extract_if(|i| {
                             self.phone_packages[*i_user][*i].state != PackageState::Enabled
                         });
                     }
                     Action::Restore => {
-                        selected_packages.drain_filter(|i| {
+                        selected_packages.extract_if(|i| {
                             self.phone_packages[*i_user][*i].state == PackageState::Enabled
                         });
                     }
@@ -353,7 +353,7 @@ impl List {
                     }
                     self.selection
                         .selected_packages
-                        .drain_filter(|s_i| *s_i == i);
+                        .extract_if(|s_i| *s_i == i);
                     Self::filter_package_lists(self);
                 }
                 Command::none()
diff --git a/src/main.rs b/src/main.rs
index d42af23..9f1d170 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,5 +1,5 @@
 #![windows_subsystem = "windows"]
-#![feature(drain_filter)]
+#![feature(extract_if)]
 #[macro_use]
 extern crate log;
 
